src/trigger.py

```diff
@@ -82,25 +45,78 @@ class cmd_run(Command):
     def __init__(self, sub_parser, name):
         super().__init__(sub_parser, name)
         self._logger = Logger("config/logger.conf", "trigger")
+        self._build_configs = None
+        self._db = None
+
+    def _log_revision(self, message, build_config, head_commit):
+        self._logger.log_message(
+            logging.INFO,
+            f"{message:32s} {build_config.name:32s} {head_commit}"
+        )
+
+    def _run_trigger(self, build_config, force):
+        head_commit = kernelci.build.get_branch_head(build_config)
+        node_list = self._db.get_nodes({
```

Comment:
As we just need to check if the revision node exists or not, this can be done by getting the count instead of the actual node.
```suggestion
        node_list = self._db.count_nodes({
```

Reply:
I think that can be fixed as a follow-up PR.  The count API entry point wasn't available when this PR was made.